cobc/scanner.l

```diff
@@ -141,6 +141,19 @@ struct cb_top_level_78 {
 	struct cb_level_78	*lev_78_ptr;
 };
 
+enum cb_literal_type {
```

Comment:
I'm fine with the internal refactoring to use the literal type enum, feel free to commit that along with a Changelog entry to svn, which will reduce the changes in this PR a bit.